src/misc/armorstand-pose.ts

```diff
@@ -0,0 +1,179 @@
+  rightLeg: new EulerAngle(0, 0, 0.2),
+};
+// prettier-ignore
+const POSE_HAND_RAISED = {
+    head:     new EulerAngle(0, 0, 0),
+    leftArm:  new EulerAngle(-3, 0, 2.5),
+    rightArm: new EulerAngle(0, 0, 2.5),
+    leftLeg:  new EulerAngle(0, 0, -0.2),
+    rightLeg: new EulerAngle(0, 0, -0.2),
+  };
+// prettier-ignore
+const POSE_ZOMBIE = {
+    head:     new EulerAngle(0, 0, 0),
+    leftArm:  new EulerAngle(-1.57, 0, -1.57),
+    rightArm: new EulerAngle(-1.57, 0, 1.57),
+    leftLeg:  new EulerAngle(0.1, 0, -0.1),
+    rightLeg: new EulerAngle(-0.1, 0, 0.1),
+};
+// prettier-ignore
+const POSE_RUNNING = {
+    head:     new EulerAngle(0, 0, 0),
+    leftArm:  new EulerAngle(-2, 0, -0.5),
+    rightArm: new EulerAngle(1, 0, 0.5),
+    leftLeg:  new EulerAngle(1, 0, -0.2),
+    rightLeg: new EulerAngle(-1, 0, 0.2),
+};
+// prettier-ignore
+const POSE_STICK = {
+    head:     new EulerAngle(1, 0, 0),
+    leftArm:  new EulerAngle(3, 0, -3),
+    rightArm: new EulerAngle(3, 0, 3),
+    leftLeg:  new EulerAngle(3, 0, -3),
+    rightLeg: new EulerAngle(3, 0, 3),
+};
+// prettier-ignore
+const POSE_LAUNCH = {
+    head:     new EulerAngle(-1, 0, 0),
+    leftArm:  new EulerAngle(2, 0, -2),
+    rightArm: new EulerAngle(2, 0, 2),
+    leftLeg:  new EulerAngle(2, 0, -2),
+    rightLeg: new EulerAngle(2, 0, 2),
+};
+// prettier-ignore
+const POSE_SITTING = {
+    head:     new EulerAngle(-0.1, 0, 0),
+    leftArm:  new EulerAngle(0, 0, -0.1),
+    rightArm: new EulerAngle(0, 0, 0.1),
+    leftLeg:  new EulerAngle(-1.6, 0, 1.6),
+    rightLeg: new EulerAngle(-1.6, 0, -1.6),
+};
+
+// prettier-ignore
+const POSES = [
+    POSE_X,
+    POSE_HUG,
+    POSE_HAND_RAISED,
+    POSE_ZOMBIE,
+    POSE_RUNNING,
+    POSE_STICK,
+    POSE_LAUNCH,
+    POSE_SITTING
+]
+
+// Change the pose of the armorstand
+registerEvent(PlayerInteractAtEntityEvent, (event) => {
+  const entity = event.rightClicked;
+  if (entity.type !== EntityType.ARMOR_STAND) return;
+  const inventory = event.player.inventory as PlayerInventory;
+  if (!inventory.itemInMainHand.type.isEmpty()) return;
+  if (!inventory.itemInOffHand.type.isEmpty()) return;
+  const armorstand = entity as ArmorStand;
+  if (!canBeChanged(armorstand)) return;
+  if (hasItems(armorstand)) return;
+
+  event.setCancelled(true);
+
+  let pose = POSES[Math.floor(Math.random() * POSES.length)];
+
+  // Small chance to get completely random pose
+  if (Math.random() < 0.2) {
+    pose = {
+      head: randomEulerAngle(),
+      leftArm: randomEulerAngle(),
+      rightArm: randomEulerAngle(),
+      leftLeg: randomEulerAngle(),
+      rightLeg: randomEulerAngle(),
+    };
+  }
+
+  armorstand.headPose = pose.head;
+  armorstand.leftLegPose = pose.leftLeg;
+  armorstand.rightLegPose = pose.rightLeg;
+  armorstand.leftArmPose = pose.leftArm;
+  armorstand.rightArmPose = pose.rightArm;
+});
+
+// Add hands for armorstand (Click with a stick)
+registerEvent(PlayerInteractAtEntityEvent, (event) => {
```

Comment:
Can the arms be removed without breaking the armor stand? Is it ok for sticks to disappear when it is broken?